### DISCUSSION TITLE
Github non-code files: using Verovio to render music files

### DISCUSSION THREAD

**Discussion by raiatea** [rism-digital/verovio#3656] [General]
Hello,

I'm thinking about a multi-person effort to typeset music score in some format (MusicXML or MEI).
Maybe a site like GitHub would be helpful if it can render music scores.
What do you think about adding that support using the Verovio library ?
I don't know if and how Github is organized to add such support.

**Comment by craigsapp** [comment]
Github works well for collaborative editing of scores (provided that all users understand how to use it).

Check out the Chopin Institute's github repository of about 8000 scores:
https://github.com/pl-wnifc/humdrum-polish-scores

Frontend that uses verovio to display the scores: https://polishscores.org

This repository is actually the publication repository, and there is a separate repository on their internal [gitea](https://about.gitea.com/) repositories, but essentially it is the same a github.

Accessing the digital scores from github is done with the "raw" data.  For example, here is digital score viewed on the github web interface:

https://github.com/pl-wnifc/humdrum-polish-scores/blob/main/pl-wtm/kern/pl-wtm--1000-sp-5590_szymanowski-karol--daleko-zostal-caly-swiat-op-2-nr-1.krn

There is a "raw" button on the top right side of the page:

<img width="485" alt="Screenshot 2024-04-26 at 12 17 40 AM" src="https://github.com/rism-digital/verovio/assets/3487289/c98367a2-ef69-46dc-bb73-9659bc672e37">

If you click on that, you have access to the "raw" contents of the file:

https://raw.githubusercontent.com/pl-wnifc/humdrum-polish-scores/main/pl-wtm/kern/pl-wtm--1000-sp-5590_szymanowski-karol--daleko-zostal-caly-swiat-op-2-nr-1.krn

With such access to the raw file's contents, you can load into JavaScript (or other programming interface) and then render with verovio, as is done on the polishscores.org website (view this score's [work page](https://polishscores.org/?id=pl-wtm--1000-sp-5590)), or into [Verovio Humdrum Viewer](https://verovio.humdrum.org/?file=https://raw.githubusercontent.com/pl-wnifc/humdrum-polish-scores/main/pl-wtm/kern/pl-wtm--1000-sp-5590_szymanowski-karol--daleko-zostal-caly-swiat-op-2-nr-1.krn) for online editing.

**Reply by raiatea** [reply]
Very informative, thank you

**Comment by ahankinson** [comment]
Hold on... I think what @raiatea is asking is slightly different. They are asking if GH can be approached to add Verovio support for their file viewers, similar to that for CSV and GeoJSON: 

 - https://github.com/smartchicago/chicago-atlas/blob/master/db/import/zipcodes.geojson
 - https://github.com/Caged/portland-census/blob/6c71d5317fb0fb36c78068b8fcb8e59d69f86bad/source/data/2010-census.csv

With a quick look I found: https://github.com/orgs/community/discussions

Which seems to be where you can file feature requests. @raiatea , would you be willing to open a discussion topic over there? If you do and report back, we could respond over there as well.

**Comment by craigsapp** [comment]
Yes that would be a very good idea.

**Comment by raiatea** [comment]
Both answers are relevant, thank you.
Github can support non-code files, Verovio provides all the tools, but polishscores answers my question, and I'm not sure Github should support any domain just because it has collaborative capabilities.
